soci/store/store.go

```diff
@@ -334,7 +334,12 @@ func (s *ContainerdStore) Push(ctx context.Context, expected ocispec.Descriptor,
 		return fmt.Errorf("unexpected copy size %d, expected %d: %w", totalWritten, expected.Size, errdefs.ErrFailedPrecondition)
 	}
 
-	return writer.Commit(ctx, expected.Size, expected.Digest)
+	err = writer.Commit(ctx, expected.Size, expected.Digest)
+	if err != nil && errors.Is(err, errdefs.ErrAlreadyExists) {
```

Comment:
nit: For consistency, I think we tend to follow the pattern

```go
if err != nil && !errors.Is(err, errdef.ErrAlreadyExists) {
  return err
}
```

Reply:
tbh I have seen all kinds of error check patterns across the codebase, but yeah this one is the most used. I will update to follow this.